proof/refine/ARM/SchedContextInv_R.thy

```diff
@@ -640,42 +639,26 @@ lemma updateRefillHd_corres:
        apply (clarsimp simp: sc_relation_def)
        apply (case_tac "sc_refills sc = []")
         apply clarsimp
-       apply (intro conjI impI)
         apply (prop_tac "scRefills sc' \<noteq> []")
          apply clarsimp
         apply (rule nth_equalityI)
-         apply (fastforce simp: refills_map_def length_replaceAt)
+         apply (fastforce simp: refills_map_def)
 
         apply (case_tac "i=0")
          apply (clarsimp simp: refills_map_def)
          apply (subst hd_map, fastforce)
          apply (subst hd_wrap_slice; simp)
-         apply (subst nth_map)
-          apply (subst length_wrap_slice; simp?)
-          apply (metis length_replaceAt)
```

Comment:
Do you have a sense of which change allows you to remove all these bits? I'm not complaining, just curious. :-)

Reply:
It’s one of the consequences of putting `length_replaceAt` in the simplest.

I think it is safe to put `length_xxx` lemmas that prove "function `xxx` preserves the size" in the simp set.